storage_backend_s3/components/s3_adapter.py

```diff
@@ -24,15 +24,20 @@ class S3StorageAdapter(Component):
     _usage = "amazon_s3"
 
     def _aws_bucket_params(self):
+
         params = {
             "aws_access_key_id": self.collection.aws_access_key_id,
             "aws_secret_access_key": self.collection.aws_secret_access_key,
-            "region_name": self.collection.aws_region,
         }
         if self.collection.aws_host:
             params["endpoint_url"] = self.collection.aws_host
-            # region must be excluded, otherwise endpoint is ignored
-            params.pop("region_name", None)
+
+        if self.collection.aws_region == 'other' and \
+                self.collection.aws_other_region:
+            params['region_name'] = self.collection.aws_other_region
+        elif self.collection.aws_region:
```

Comment:
if we don't have an empty option - as it seems to be - is going to break non AWS services.

Reply:
AFAIU the region is needed if you rely on AWS routing, hence IMO you don't need a region if you specify a custom endpoint -> that's why I popped the region. 
Have you experienced something else?

Reply:
Im my example above, you can see that you can choose other region and keep the region name empty. 

Because not all non-AWS services have no defined region.

Your code don't work with the code example above.

Reply:
`aws_region` cannot be empty, look at your radio widget ;)
You are forced to select something. You should add `[('', 'None')]` to the options.

Reply:
also, I added a test for this: can you update it or add another one?

Reply:
You are right sorry, i fixed the logic and push again. 

Please take a look.